tabby-ssh/src/components/sftpCreateDirectoryModal.component.ts

```diff
@@ -0,0 +1,38 @@
+import { Component } from '@angular/core'
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap'
+import path from 'path'
+import { BaseComponent } from 'tabby-core'
+import { SFTPFile, SFTPSession } from '../session/sftp'
+import { SFTPPanelComponent } from './sftpPanel.component'
+
+/** @hidden */
+@Component({
+    template: require('./sftpCreateDirectoryModal.component.pug'),
+})
+export class SFTPCreateDirectoryModalComponent extends BaseComponent {
```

Comment:
Some nitpicks on the architecture: I'd prefer if `SFTPCreateDirectoryModalComponent` just returned (via `modalInstance.close`) the new name, and `SFTPPanelComponent` would then handle actual creation of the folder. This way you don't have to send the `SSHSession` around.

Reply:
should be better now :)